### DISCUSSION TITLE
is the "first-step" notebook tested for the current main branch?

### DISCUSSION THREAD

**Discussion by Srheft** [eradiate/eradiate#332] [Q&A] [answered]
Hi Eradiate,

New user question:
My installation seemed to pass the sanity check but main branch fails to run [eradiate_quickstart.ipynb](https://github.com/eradiate/eradiate-tutorials/blob/main/getting_started/eradiate_quickstart.ipynb)
seems like 'spectral_cfg' is outdated for the most up-to-date main branch but the notebook uses it. 

when I switch to f943b96bc6c7b3baff32ac6c4e99802606cff85c , I can run the notebook but the radiance values are all zero. 

is it possible for an experienced user to share a working example of 'measure'  or for developing team to update the notebook?

Many thanks,
S.

**Comment by nollety** [comment]
Hi Sarah,

Thank you for your question.
You are right: `spectral_cfg` was replaced by the `srf` parameter and is outdated now. What version of the documentation are you using? (visible in the bottom left of the page on Read The Docs)
I recommend that you check out the [`latest`](https://eradiate.readthedocs.io/en/latest/) version (provided that you installed also the latest Eradiate version).

Here is a working example for the measure definition:

```python
import numpy as np

import eradiate.scenes as ertsc
from eradiate import unit_registry as ureg

measure = ertsc.measure.MultiDistantMeasure.hplane(
    id="toa_brf",
    zeniths=np.arange(-75, 76, 5),
    azimuth=0,
    srf={"type": "multi_delta", "wavelengths": 550.0 * ureg.nm},
    spp=10000,
) 
```

**Comment by Srheft** [comment]
Thank you for your reply, Yvan!
 I just pulled the latest changes in 'main' branch and saw the `srf` change in the notebook. I had been looking at the web version of the notebook [linked in my original message].

Now,  although `measure` is in acceptable format, seems like `eradiate.experiments.AtmosphereExperiment()` does not find parameter `w` in the dataset:

Running the cell right after defining measure in quickstart notebook, I get this error message below:

```python
exp = eradiate.experiments.AtmosphereExperiment(
    surface=surface_bsdf,
    atmosphere=atmosphere,
    illumination=illumination,
    measures=measure,
)
```


```
File ~/eradiate/src/eradiate/spectral/mono.py:71, in WavelengthSet.from_absorption_dataset(cls, dataset)
     57 Create a wavelength set from an absorption dataset.
     58 
   (...)
     67     Generated wavelength set.
     69 # we dont know if the absorption dataset 'w' variable holds
     70 # wavelengths or wavenumbers, so we check the units
---> 71 w = to_quantity(dataset.w)
     72 if w.check("[length]^-1"):
     73     wavelengths = np.sort(1 / w).to(
     74         "nm"
     75     )  # ordered wavenumbers are reversed in wavelength-space

AttributeError: 'NoneType' object has no attribute 'w'
```

Appreciate any hint as to what could be wrong here. Thanks, S.

**Comment by nollety** [comment, marked as answer]
This is due to the fact that, in monochromatic mode, the atmosphere expects an absorption dataset (it is not loaded by default, like it used to be previously).
This inconvenience is an issue that was introduced recently and that should be fixed in the coming days, so that the atmosphere can be instantiated without any required parameter in both CKD and monochromatic modes.

At the moment, however, the following will lead to `Experiment` failing (as you experienced):
```python
atmosphere = ertsc.atmosphere.MolecularAtmosphere.ussa_1976()
```

Instead, you should locate and open the monochromatic absorption dataset corresponding to the processed spectral interval:
```python
import pathlib

from eradiate import data

path = pathlib.Path("spectra/absorption/us76_u86_4/us76_u86_4-spectra-18000_19000.nc")  # spans the [18000, 19000] cm^-1 wavenumber range
dataset = data.open_dataset(path)
```
and pass it to the atmosphere constructor:
```python

atmosphere = ertsc.atmosphere.MolecularAtmosphere.ussa_1976(absorption_dataset=dataset)
```

Note that the U.S. 1976 standard atmosphere does not include water vapour nor ozone so there is hardly no atmospheric absorption at 550 nm. This is why the tutorial uses:
```python
atmosphere = ertsc.atmosphere.MolecularAtmosphere.ussa_1976(has_absorption=False)
```
which makes `absorption_dataset` not required.

Thank you for your feedback about using Eradiate, this is really valuable to us and I'll be glad to help further if you have other questions.

**Reply by Srheft** [reply]
Thank you for the explanation. Both when using the dataset and atmosphere or setting `has_absorption=False`, I can generate the `results` with null `brf` (I had thought to be able to see reflectance as shown in the first step notebook - switching to the commit mentioned in original message, I'd get null brf as well which made me question the standard execution of experiment in my  local installation). 

![Screenshot 2023-06-07 at 8 53 52 AM](https://github.com/eradiate/eradiate/assets/32471902/3e8cf222-a575-45cd-be4c-2d5634d3c791)

**Reply by nollety** [reply]
Is there a particular reason for choosing [f943b96](https://github.com/eradiate/eradiate/commit/f943b96bc6c7b3baff32ac6c4e99802606cff85c)? What if you checkout the latest main commit? (you can find the instructions in the [update guide](https://eradiate.readthedocs.io/en/latest/rst/getting_started/update.html))
What output do you get when you enter `eradiate show` in a terminal (with the eradiate conda environment active)?

**Reply by Srheft** [reply]
Initially, when I was using the outdated notebook, and `spectral_cfg` failure was happening, I switched to older commits to get to a version 
 that worked with `spectral_cfg`. that was basically the reason for choosing that commit.
Ever since our exchange, I have been using the main branch with latest changes I pulled. This is the output of `eradiate show`:

```
── System ──────
CPU: b'Apple M2 Pro'
OS: 13.4
Python: 3.8.16 | packaged by conda-forge | (default, Feb  1 2023, 16:05:36) [Clang 14.0.6 ]
── Versions ─────────
• eradiate 0.23.1.post1.dev26+g9a3bdcee
• drjit 0.4.1
• mitsuba 3.2.1
── Available Mitsuba variants ──
• scalar_mono
• scalar_mono_double
• scalar_rgb
• scalar_spectral
• llvm_mono_double
• llvm_ad_rgb
── Configuration ─────────
• ERADIATE_AZIMUTH_CONVENTION: AzimuthConvention.EAST_RIGHT
• ERADIATE_DATA_STORE_URL: http://eradiate.eu/data/store/
• ERADIATE_DOWNLOAD_DIR: /Users/seftekh1/eradiate/resources/downloads
• ERADIATE_OFFLINE: False
• ERADIATE_PROGRESS: ProgressLevel.KERNEL (2)
• ERADIATE_SOURCE_DIR: /Users/seftekh1/eradiate
```

**Reply by nollety** [reply]
I am not sure if we support Apple M2 CPU? @leroyvn

**Reply by leroyvn** [reply]
In principle, we do, yes: Mitsuba builds on M2. But we haven't tested it.

**Reply by Srheft** [reply]
Thank you for the info. For clarity, I just add that all the steps in update guide were completed successfully and I am using the latest main branch that outputs the `results` with null `brf`. I have access to a mac intel CPU that i could try and report back. if anything came to mind for me to try on the M2 system, please let me know.

**Reply by nollety** [reply]
It would be interesting to see the results on a Mac Intel CPU :+1:  (I don't have access to an Apple M2 at the moment)

**Reply by Srheft** [reply]
Good news!  using the Mac Intel CPU system, I was able to get`brf`.
The only issue is that this device is not mine and I only borrowed it to test; still hoping to find a solution for the M2 system. Much appreciate your swift replies to this thread and I hope it'll be useful for other users. 

<img width="730" alt="image" src="https://github.com/eradiate/eradiate/assets/32471902/0ba67429-b794-4e24-aff8-cdd66fe9cd17">

**Reply by nollety** [reply]
Good to know, many thanks for reporting that!
